src/main/java/teammates/common/util/AppUrl.java

```diff
@@ -0,0 +1,45 @@
+package teammates.common.util;
+
+/**
+ * The specific implementation of {@link Url} used to encapsulate URLs of the application.
+ * The base URL used in the conversion is the value of app.url in build.properties.
+ */
+public class AppUrl extends Url {
+
+    public AppUrl(String url) {
+        this(Config.APP_URL, url);
+    }
```

Comment:
I just thought of something. If let's say I make a factory method `Config.getAppUrl(...)` then it will make `Config` dependent to the constructor of `AppUrl`. How is it any better than `AppUrl` being dependent to just one static field of `Config`?
I don't think testability is an issue, because `AppUrl` is just a specific implementation of `Url` without any new method so as long as `Url` is fully tested, then so is `AppUrl`.
This design of `Url` and its subclasses is actually similar to `FeedbackQuestionDetails` and its subclasses, except that `Url` is not abstract and I can make it abstract if it serves its purpose better.